packages/sku/lib/SkuConfigUpdater.js

```diff
@@ -0,0 +1,186 @@
+      let configAst;
+
+      log(
+        'No default export found in sku config. Config is either CJS or invalid.',
+      );
+
+      t.assertProgram(skuConfigModule.$ast);
+      const lastStatement = skuConfigModule.$ast.body.at(-1);
+      t.assertExpressionStatement(lastStatement);
+
+      t.assertAssignmentExpression(lastStatement.expression);
+      const { expression } = lastStatement;
+      t.assertMemberExpression(expression.left);
+      t.assertIdentifier(expression.left.object, {
+        name: 'module',
+      });
+      t.assertIdentifier(expression.left.property, {
+        name: 'exports',
+      });
+
+      if (t.isObjectExpression(expression.right)) {
+        configAst = expression.right;
+      } else if (t.isIdentifier(expression.right)) {
+        const skuConfigIdentifierName = expression.right.name;
+        const skuConfigDeclaration = skuConfigModule.$ast.body.find(
+          (node) =>
+            t.isVariableDeclaration(node) &&
+            t.isIdentifier(node.declarations[0].id) &&
+            node.declarations[0].id.name === skuConfigIdentifierName,
+        );
+        assert(skuConfigDeclaration, 'Expected skuConfig to be defined');
+        t.assertVariableDeclaration(skuConfigDeclaration);
+        t.assertVariableDeclarator(skuConfigDeclaration.declarations[0]);
+        t.assertObjectExpression(skuConfigDeclaration.declarations[0].init);
+        configAst = skuConfigDeclaration.declarations[0].init;
+      } else {
+        throw new Error("Couldn't find config object in CJS sku config");
+      }
+
+      this.#config = {
+        type: 'cjs',
+        configAst,
+      };
+    } else {
+      log('Found sku config with ESM export');
+
+      if (skuConfigModule.exports.default.$type === 'object') {
+        const configAst = skuConfigModule.exports.default;
+        this.#config = {
+          type: 'esm',
+          configAst,
+        };
+      } else {
+        const { declaration: configDeclaration, config: configAst } =
+          getConfigFromVariableDeclaration(skuConfigModule);
+
+        assert(configAst, 'Expected skuConfig to be defined');
+        this.#config = {
+          type: 'esm-non-literal',
+          configAst,
+          configDeclaration,
+        };
+      }
+    }
+  }
+
+  /**
+   * Creates a new instance of SkuConfigUpdater from a file path
+   *
+   * @param {string} path - An absoulte path to a sku config
+   */
+  static async fromFile(path) {
+    const contents = await readFile(path, 'utf8');
+
+    return new SkuConfigUpdater({ path, contents });
+  }
+
+  /**
+   * Updates `property` in sku config with the provided `value`. Inserts the `property` and `value` if it doesn't exist.
+   *
+   * This method does not write the changes to the file system. Use `commitConfig` to do that.
+   *
+   * @template {keyof SkuConfig} T
+   * @param {{property: T, value: SkuConfig[T]}} options
+   */
+  upsertConfig({ property, value }) {
+    if (this.#config.type === 'cjs') {
```

Comment:
**nitpick:** worth creating constants for these string values?

typo's can happen since this is plain JS.